### DISCUSSION TITLE
[5] Dinder - First Programming

### DISCUSSION THREAD

**Discussion by Manuel-F-04** [dhbw-ka-tinf22b5-dinder/Dinder#10] [Projektblog]
### Frontend:
We built a simple login and registration form and a front page.
We also created a navigation bar with a logo and implemented routing.
When you click on the logo, you come back to the front page.

![registration page](https://github.com/dhbw-ka-tinf22b5-dinder/Dinder/assets/48451580/ab67d0de-7802-43b0-a1b9-4726ecb89d84)
![front page](https://github.com/dhbw-ka-tinf22b5-dinder/Dinder/assets/48451580/64bf0eb9-4ee2-45a3-869e-56c51775190b)
![login page](https://github.com/dhbw-ka-tinf22b5-dinder/Dinder/assets/48451580/faab15d1-acf7-4b95-b3e6-8e093ee1b3ca)

There is no logic behind it at the moment because React is relatively new to us
and we had to spend a lot of time learning it.

### Backend:
We have implemented simple cookie authentication for API calls, 
as well as a simple login interface and integrated the database. 

The database was changed from MariaDB to PostgreSQL due to available free hosts, 
and the database host is Supabase. The first draft for the individual tables 
was also created for the database.

### Distribution of tasks:
Tobias - Backend (Authentication, Authorization)
Jonathan - Backend (login interface)
Manuel - Database (find host, create table schema)
Jannis - Frontend (get to know React, login - Page)

**Comment by ghost** [comment]
Das klingt sehr interessant. Besonders gut fand ich euren Wechsel von MariaDB zu PostgreSQL und der damit verbundene Wechsel zu Supabase als gratis Hostingservice. Ebenfalls toll zu sehen ist, dass ihr bereits eine Authentifizierung für eure API gebaut habt.

Weiter so.

**Comment by Puggingtons** [comment]
Euer Frontend sieht schon super aus. Die Implementierung einer Authentifizierung ist auch schon ein guter Fortschritt :)

**Comment by lukas-mchr** [comment]
Schöne Gestaltung der Login- und Registration-Forms. Auch schön zu sehen, dass ihr euch bereits um die Authentifizierung kümmert.

**Comment by DossohMael** [comment]
Sehr guter Fortschritt, besonders hat mir gefallen dass ihr nun eine Authentifizierung für die API eingebaut habt! 
Sicherheit geht vor!
